tests/unit/Rules/NegativeTest.php

```diff
@@ -13,65 +13,53 @@
 
 namespace Respect\Validation\Rules;
 
-use PHPUnit\Framework\TestCase;
+use Respect\Validation\Test\RuleTestCase;
 
 /**
- * @group  rule
+ * @group rule
+ *
  * @covers \Respect\Validation\Rules\Negative
- * @covers \Respect\Validation\Exceptions\NegativeException
+ *
+ * @author Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ * @author Gabriel Caruso <carusogabriel34@gmail.com>
+ * @author Henrique Moody <henriquemoody@gmail.com>
+ * @author Ismael Elias <ismael.esq@hotmail.com>
  */
-class NegativeTest extends TestCase
+final class NegativeTest extends RuleTestCase
 {
-    protected $negativeValidator;
-
-    protected function setUp(): void
-    {
-        $this->negativeValidator = new Negative();
-    }
-
-    /**
-     * @dataProvider providerForNegative
-     */
-    public function testNegativeShouldPass($input): void
+    /*
+    * {@inheritdoc}
+    */
+    public function providerForValidInput(): array
     {
-        $this->negativeValidator->assert($input);
-        self::assertTrue($this->negativeValidator->__invoke($input));
-        $this->negativeValidator->check($input);
-    }
-
-    /**
-     * @dataProvider providerForNotNegative
-     * @expectedException \Respect\Validation\Exceptions\NegativeException
-     */
-    public function testNotNegativeNumbersShouldThrowNegativeException($input): void
-    {
-        self::assertFalse($this->negativeValidator->__invoke($input));
-        $this->negativeValidator->assert($input);
-    }
+        $rule = new Negative();
 
-    public function providerForNegative()
-    {
         return [
-            ['-1.44'],
-            [-1e-5],
-            [-10],
+            [$rule, '-1.44'],
+            [$rule, -1e-5],
+            [$rule, -10],
         ];
     }
 
-    public function providerForNotNegative()
+    /*
+    * {@inheritdoc}
+    */
+    public function providerForInvalidInput(): array
     {
+        $rule = new Negative();
+
         return [
-            [''],
-            [0],
-            [-0],
-            [null],
-            ['a'],
-            [' '],
-            ['Foo'],
-            [16],
-            ['165'],
-            [123456],
-            [1e10],
+            [$rule, ''],
```

Comment:
Because this PR is adding a new behaviour it should have added a few more tests, and updates the changelog of the rule.